### PR TITLE
Use `benchmarking::v2` in collator selection pallet

### PR DESCRIPTION
**Pull request by georgepisaltu**
Move from `frame_benchmarking::v1` to `frame_benchmarking::v2` in `pallet-collator-selection`.

### CODE REVIEW

File: pallets/collator-selection/src/benchmarking.rs [hunk 1]
```diff
@@ -119,31 +112,41 @@ fn min_invulnerables<T: Config>() -> u32 {
...
 		sorted_new_invulnerables.sort();
-	}: {
-		assert_ok!(
-			// call the function with the unsorted list
-			<CollatorSelection<T>>::set_invulnerables(origin, new_invulnerables.clone())
-		);
-	}
-	verify {
+
+		#[block]
```
[the remark this answers is not in the document]
Comment (georgepisaltu) [reply pallets/collator-selection/src/benchmarking.rs:130]: I tried the `#[extrinsic_call]` syntax but it doesn't work with the associated `assert`. As far as I understand, if it's more than just the extrinsic call itself, it needs to be a `#[block]`.

[the remark this answers is not in the document]
Comment (georgepisaltu) [reply pallets/collator-selection/src/benchmarking.rs:130]: I was wrong before then, the `assert` is to make the `#[block]` equivalent to `#[extrinsic_call]`. The reason it doesn't work with `#[extrinsic_call]` is the `origin` variable, which is a type mismatch when paired with `#[extrinsic_call]`. I've been trying to solve this type mismatch for a while now but I couldn't so far. More specifically, using `T::UpdateOrigin::try_successful_origin()` in this case when in the other `#[extrinsic_call]`s `RawOrigin` is used.

[the remark this answers is not in the document]
Comment (ggwpez) [reply pallets/collator-selection/src/benchmarking.rs]: There is a custom syntax for that 😄 I fixed it.

[the remark this answers is not in the document]
Comment (georgepisaltu) [reply pallets/collator-selection/src/benchmarking.rs]: Done. PTAL and thanks for the help!

**Review by ggwpez — APPROVED** [review]
Bueno!

Please let me know if you encounter other problems with the V2 benchmarking syntax in the future - then we can fix them.

**Review by joepetrowski — APPROVED** [review]

**Comment by georgepisaltu** [comment]
bot merge

**Pull request merged**